staging/src/k8s.io/kube-aggregator/pkg/controllers/autoregister/autoregister_controller.go

```diff
@@ -58,9 +59,9 @@ type AutoAPIServiceRegistration interface {
 	RemoveAPIServiceToSync(name string)
 }
 
-// autoRegisterController is used to keep a particular set of APIServices present in the API.  It is useful
+// Controller is used to keep a particular set of APIServices present in the API.  It is useful
 // for cases where you want to auto-register APIs like TPRs or groups from the core kube-apiserver
-type autoRegisterController struct {
+type Controller struct {
```

Comment:
we shouldn't export this type just to make lint happy

Reply:
Across the codebase, most controllers similar to this are exported.
Is there another way to satisfy Golint, or will I have to leave this package in `.golint_failures`?

Reply:
@liggitt I agree with @eloyekunle. Let's not block this by making Controller public. We need a better golint exception mechanism.

Reply:
> I agree with @eloyekunle. Let's not block this by making Controller public

not sure if you are suggesting keeping this package in .golint_failures or making Controller public. I'd prefer not to export just to make golint happy, but I don't feel that strongly

Reply:
Jordan, why is it important for this to stay private?

Automated systems like this and gofmt don't always produce the ideal decision at the micro level (maybe you hate tabs, or disagree with lint's exported type directives), but at the macro level the value of providing consistent expectations makes it easier for large teams to collaborate on large code bases.

Removing this package from .golint_failures is useful for two reasons:
1) It ensures no other, potentially more useful failures, get checked into this package.
2) A shorter .golint_failure exception lint makes it more likely other packages will fix their lint errors rather than being one of the few special-cased packages.

IMO it is generally an unproductive use of time to try and fight the linting system (but maybe this is an exceptional circumstance?)

Reply:
> why is it important for this to stay private?

this is a package in an exported repo, so making the controller class public implies it is a suitable API for other projects to depend on. I don't think its structure has been considered in that light, so simply making it public to pass a linting check seems subpar

Reply:
Jordan, is this the only controversial part about the PR?

Elijah, if so how would you feel about removing this part from this PR. This will expedite removing most of the lint errors. 

If you're up for it, I'd suggest creating a second PR with just the autoRegisterController change (and removing the lint exception). We can continue the discussion there.

Reply:
yes, everything else looked fine to me

Reply:
Sounds good, I've removed this part from the PR. The controller is now unexported.